### ISSUE TITLE
Game freezes on loading screen (1.20.1)

### ISSUE THREAD

**Issue by DevLung** [Chocohead/OptiFabric#1105]
I wanted to use optifabric with some quality-of-life mods I use (screenshot of all mods below) (I use fabric loader 0.14.21 for 1.20.1)
![Bild_2023-07-14_193437563](https://github.com/Chocohead/OptiFabric/assets/123473305/15ec6c15-a17f-4ed4-b942-d2e50216a6c6)
but when starting the game, it freezes on the loading screen, but doesn't crash. The application is still responding and I can close it as usual but it just doesn't do anything...
![Bild_2023-07-14_193824216](https://github.com/Chocohead/OptiFabric/assets/123473305/a9897575-b3df-4cb0-b09b-ef8d42e6ff92)

I tried troubleshooting by removing all the mods except for optifabric and OptiFine (even removed fabric api) and the game started.
I then added fabric api back in and it worked.
Adding shulkerbox tooltip or modmenu made it freeze again but everything else seemed to work.
Somehow I then got it all working again (idk how)
But now I wanted to start the instance again and it froze: tried to do the same steps again but now it ONLY starts with I just optifabric and OptiFine.
I even reinstalled fabric from the newest installer and re-downloaded all the mods hoping there had been some corrupt/buggy files but that didn't help either

**TL;DR** The game sometimes freezes and sometimes starts normally in multiple different configurations without a clear reason.

btw all the mods work perfectly together without optifabric installed and I used these mods before with optifabric in 1.19.3 and it worked without problems.
The game even starts with all the mods + optifabric without the optifine jar (but obviously gives the "no optifine jar found" error on startup)
optifabric sometimes just seems to fail to fully load optifine if any of these other mods are loaded/loading.

**Comment by DevLung** [comment]
Tried it again just now and after restarting the game 3 times it actually started but that's obviously not a solution, there is definitely some bug going on...

**Comment by d4v1ddr0dr1gu3s** [comment]
Same

**Comment by solvanan** [comment]
same

**Comment by darkejon** [comment]
same here too
Error: 

Incompatible mod set!
net.fabricmc.loader.impl.FormattedException: Mod resolution encountered an incompatible mod set!
A potential solution has been determined:
	 - Replace mod 'OptiFabric' (optifabric) 1.13.16 with any version that is compatible with:
		 - minecraft 1.20.1
Unmet dependency listing:
	 - Mod 'OptiFabric' (optifabric) 1.13.16 requires version 1.16.1, version 1.16.2, version 1.16.3, version 1.16.4, version 1.16.5, version 1.17-alpha.21.8.b, version 1.17-beta.4, version 1.17, version 1.17.1, version 1.18-beta.1, version 1.18, version 1.18.1, version 1.18.2, version 1.19, version 1.19.1 or version 1.19.2 of 'Minecraft' (minecraft), but only the wrong version is present: 1.20.1!
	at net.fabricmc.loader.impl.FormattedException.ofLocalized(FormattedException.java:51)
	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:192)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:148)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)

**Comment by AuroraDoesCode** [comment]
> same here too Error:
> 
> Incompatible mod set! net.fabricmc.loader.impl.FormattedException: Mod resolution encountered an incompatible mod set! A potential solution has been determined: - Replace mod 'OptiFabric' (optifabric) 1.13.16 with any version that is compatible with: - minecraft 1.20.1 Unmet dependency listing: - Mod 'OptiFabric' (optifabric) 1.13.16 requires version 1.16.1, version 1.16.2, version 1.16.3, version 1.16.4, version 1.16.5, version 1.17-alpha.21.8.b, version 1.17-beta.4, version 1.17, version 1.17.1, version 1.18-beta.1, version 1.18, version 1.18.1, version 1.18.2, version 1.19, version 1.19.1 or version 1.19.2 of 'Minecraft' (minecraft), but only the wrong version is present: 1.20.1! at net.fabricmc.loader.impl.FormattedException.ofLocalized(FormattedException.java:51) at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:192) at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:148) at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68) at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88) at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126) at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)

You’re using an outdated version of Optifabric, that’s your issue. You should be using 1.13.25 I believe it is now.



> I wanted to use optifabric with some quality-of-life mods I use (screenshot of all mods below) (I use fabric loader 0.14.21 for 1.20.1) ![Bild_2023-07-14_193437563](https://user-images.githubusercontent.com/123473305/253638260-15ec6c15-a17f-4ed4-b942-d2e50216a6c6.png) but when starting the game, it freezes on the loading screen, but doesn't crash. The application is still responding and I can close it as usual but it just doesn't do anything... ![Bild_2023-07-14_193824216](https://user-images.githubusercontent.com/123473305/253638948-a9897575-b3df-4cb0-b09b-ef8d42e6ff92.png)
> 
> I tried troubleshooting by removing all the mods except for optifabric and OptiFine (even removed fabric api) and the game started. I then added fabric api back in and it worked. Adding shulkerbox tooltip or modmenu made it freeze again but everything else seemed to work. Somehow I then got it all working again (idk how) But now I wanted to start the instance again and it froze: tried to do the same steps again but now it ONLY starts with I just optifabric and OptiFine. I even reinstalled fabric from the newest installer and re-downloaded all the mods hoping there had been some corrupt/buggy files but that didn't help either
> 
> **TL;DR** The game sometimes freezes and sometimes starts normally in multiple different configurations without a clear reason.
> 
> btw all the mods work perfectly together without optifabric installed and I used these mods before with optifabric in 1.19.3 and it worked without problems. The game even starts with all the mods + optifabric without the optifine jar (but obviously gives the "no optifine jar found" error on startup) optifabric sometimes just seems to fail to fully load optifine if any of these other mods are loaded/loading.

Remove all your texture packs temporarily and try loading again, I believe it’s a texture related bug, as removing texture packs seems to fix it for me, and has since 1.19.3 at least.

**Comment by DevLung** [comment]
> Remove all your texture packs temporarily and try loading again, I believe it’s a texture related bug, as removing texture packs seems to fix it for me, and has since 1.19.3 at least.

Idk why but right now everything works fine but if it starts freezing again I'll try that, thank you!

**Comment by DevLung** [comment]
> > Remove all your texture packs temporarily and try loading again, I believe it’s a texture related bug, as removing texture packs seems to fix it for me, and has since 1.19.3 at least.
> 
> Idk why but right now everything works fine but if it starts freezing again I'll try that, thank you!

Had the Problem again but that fix seems to work.
just open %appdata%\.minecraft\options.txt (if you are on windows), find "resourcePacks" and delete everything but "vanilla" and "fabric" from the list.
After that, the game started just fine for me and I could add the packs back in.

**Comment by AuroraDoesCode** [comment]
> > > Remove all your texture packs temporarily and try loading again, I believe it’s a texture related bug, as removing texture packs seems to fix it for me, and has since 1.19.3 at least.
> > 
> > 
> > Idk why but right now everything works fine but if it starts freezing again I'll try that, thank you!
> 
> Had the Problem again but that fix seems to work. just open %appdata%.minecraft\options.txt (if you are on windows), find "resourcePacks" and delete everything but "vanilla" and "fabric" from the list. After that, the game started just fine for me and I could add the packs back in.

Yep the issue doesn’t seem to log an error, just crashes loading resource pack textures, so I’d assume there’s something going wrong somewhere in the Optifabric code in between.

**Comment by darkejon** [comment]
> You’re using an outdated version of optifine, that’s your issue. You should be using 1.13.25 I believe it is now.
> 
> >

I was using the latest beta, now using the final Optifine release: OptiFine HD U I5. Still getting the following error:

```
Incompatible mod set!
net.fabricmc.loader.impl.FormattedException: Mod resolution encountered an incompatible mod set!
A potential solution has been determined:
	 - Replace mod 'OptiFabric' (optifabric) 1.13.16 with any version that is compatible with:
		 - minecraft 1.20.1
Unmet dependency listing:
	 - Mod 'OptiFabric' (optifabric) 1.13.16 requires version 1.16.1, version 1.16.2, version 1.16.3, version 1.16.4, version 1.16.5, version 1.17-alpha.21.8.b, version 1.17-beta.4, version 1.17, version 1.17.1, version 1.18-beta.1, version 1.18, version 1.18.1, version 1.18.2, version 1.19, version 1.19.1 or version 1.19.2 of 'Minecraft' (minecraft), but only the wrong version is present: 1.20.1!
	at net.fabricmc.loader.impl.FormattedException.ofLocalized(FormattedException.java:51)
	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:192)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:148)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)

```

**Comment by AuroraDoesCode** [comment]
> > You’re using an outdated version of optifine, that’s your issue. You should be using 1.13.25 I believe it is now.
> > >
> 
> I was using the latest beta, now using the final Optifine release: OptiFine HD U I5. Still getting the following error:
> 
> ```
> Incompatible mod set!
> net.fabricmc.loader.impl.FormattedException: Mod resolution encountered an incompatible mod set!
> A potential solution has been determined:
> 	 - Replace mod 'OptiFabric' (optifabric) 1.13.16 with any version that is compatible with:
> 		 - minecraft 1.20.1
> Unmet dependency listing:
> 	 - Mod 'OptiFabric' (optifabric) 1.13.16 requires version 1.16.1, version 1.16.2, version 1.16.3, version 1.16.4, version 1.16.5, version 1.17-alpha.21.8.b, version 1.17-beta.4, version 1.17, version 1.17.1, version 1.18-beta.1, version 1.18, version 1.18.1, version 1.18.2, version 1.19, version 1.19.1 or version 1.19.2 of 'Minecraft' (minecraft), but only the wrong version is present: 1.20.1!
> 	at net.fabricmc.loader.impl.FormattedException.ofLocalized(FormattedException.java:51)
> 	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:192)
> 	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:148)
> 	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68)
> 	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
> 	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
> 	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
> 	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)
> ```

I should have corrected my comment, it’s Optifabric you need to update, not Optifine.
